src/lib.rs

```diff
@@ -237,7 +238,17 @@ impl OciDir {
     /// Open an existing OCI directory.
     pub fn open(dir: &Dir) -> Result<Self> {
         let dir = std::sync::Arc::new(dir.try_clone()?);
-        Ok(Self { dir })
+        let blobs_dir = dir.open_dir(BLOBDIR)?;
+        Ok(Self { dir, blobs_dir })
+    }
+
+    /// Open an existing OCI directory with a separate cap_std::Dir for blobs/sha256
+    /// This is useful when `blobs/sha256` might contain symlinks pointing outside the oci
+    /// directory, e.g. when sharing blobs across OCI repositories. The LXC OCI template uses this
+    /// feature.
+    pub fn open_with_external_blobs(dir: &Dir, blobs_dir: Dir) -> Result<Self> {
```

Comment:
I'm ok with this as is to be clear but it definitely feels inconsistent to have one take a ref and the other an owned value.

With just a small refactoring we could have both take owned values instead right?

Reply:
Or we take refs and do the `try_clone()` for both, if I had to make a call I'd probably say let's do that, but not strongly held.

Reply:
I actually prefer to take owned values for both. It would remove the `try_clone`, but what I really would like to avoid is [sharing the underlying file handle](https://doc.rust-lang.org/nightly/std/fs/struct.File.html#method.try_clone), where you could seek through a reference, read through the other and observe the side effect of the seek. I know I've linked the example for files, but I assume there would be similar semantics for directories. I would like to entirely avoid having to deal with such behavior.

Reply:
I guess one concern with this approach is that it's a breaking change.

Reply:
> I guess one concern with this approach is that it's a breaking change.

Yes. Now let's try to roll up a few other cleanups at least to make the semver bump worth it.